### PR TITLE
Improve docs for `DataFrame` methods

### PR DESCRIPTION
**Pull request by etiennebacher**
Improve and harmonize docs for DataFrame:

* fix typos and correct sentences
* add more info in description and parameters
* add examples or clarify existing ones

Some stuff depends on personal preferences of course but I think most changes are reasonable

### CODE REVIEW

File: R/dataframe__frame.R [hunk 1]
```diff
@@ -969,163 +1063,197 @@ DataFrame_to_struct = function(name = "") {
...
 #' df$fill_nan(99)
+
 DataFrame_fill_nan = function(fill_value) {
   self$lazy()$fill_nan(fill_value)$collect()
 }
 
-#' @title Fill null
-#' @description Fill null values using the specified value or strategy.
+#' @title Fill nulls
+#' @description Fill null values (which correspond to `NA` in a classic
```
Comment (eitsupi) [inline R/dataframe__frame.R:1216]: I don't think the word "classic" is appropriate.
Isn't it just `NA` in R and `null` in Polars?

File: R/dataframe__frame.R [hunk 2]
```diff
@@ -969,163 +1063,197 @@ DataFrame_to_struct = function(name = "") {
...
 DataFrame_slice = function(offset, length = NULL) {
   self$lazy()$slice(offset, length)$collect()
 }
 
 
-#' @title Null count
-#' @description Create a new DataFrame that shows the null counts per column.
+#' @title Count null values
+#' @description Create a new DataFrame that shows the null (which correspond
+#' to `NA` in a classic data.frame) counts per column.
```
Comment (eitsupi) [inline R/dataframe__frame.R:1256]: Ditto

**Review by eitsupi — APPROVED** [review]
Looks good! Thanks!

A few comments.

File: R/dataframe__frame.R (hunk 1 above)
Comment (etiennebacher) [reply R/dataframe__frame.R:1216]: Right, I replaced "in a classic data.frame" by "in R"

**Pull request merged**